ros/src/computing/planning/motion/packages/waypoint_follower/include/mpc_follower/mpc_follower.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018-2019 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <vector>
+#include <iostream>
+#include <limits>
+#include <chrono>
+
+#include <unistd.h>
+
+#include <ros/ros.h>
+#include <std_msgs/Float64.h>
+#include <std_msgs/Float32.h>
+#include <std_msgs/Float64MultiArray.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <visualization_msgs/Marker.h>
+#include <tf2/utils.h>
+
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/LU>
+
+#include <autoware_msgs/ControlCommandStamped.h>
+#include <autoware_msgs/Lane.h>
+#include <autoware_msgs/VehicleStatus.h>
+
+#include "mpc_follower/mpc_utils.h"
+#include "mpc_follower/mpc_trajectory.h"
+#include "mpc_follower/lowpass_filter.h"
+#include "mpc_follower/vehicle_model/vehicle_model_bicycle_kinematics.h"
+#include "mpc_follower/vehicle_model/vehicle_model_bicycle_dynamics.h"
+#include "mpc_follower/vehicle_model/vehicle_model_bicycle_kinematics_no_delay.h"
+#include "mpc_follower/qp_solver/qp_solver_unconstr.h"
+#include "mpc_follower/qp_solver/qp_solver_unconstr_fast.h"
+
+class MPCFollower
+{
+public:
+  MPCFollower();
+  ~MPCFollower();
+
+private:
+  ros::NodeHandle nh_, pnh_;
+  ros::Publisher pub_steer_vel_ctrl_cmd_, pub_twist_cmd_;
+  ros::Subscriber sub_ref_path_, sub_twist_, sub_pose_, sub_vehicle_status_;
+  ros::Timer timer_control_;
+
+  MPCTrajectory ref_traj_;                                   // reference trajectory for mpc
+  Butterworth2dFilter lpf_steering_cmd_;                     // steering command lowpass filter
+  Butterworth2dFilter lpf_lateral_error_;                    // lateral error lowpass filter
+  Butterworth2dFilter lpf_yaw_error_;                        // yaw error lowpass filter
+  autoware_msgs::Lane current_waypoints_;                    // current received waypoints
+  std::shared_ptr<VehicleModelInterface> vehicle_model_ptr_; // vehicle model for mpc
+  std::string vehicle_model_type_;                           // vehicle model type
+  std::shared_ptr<QPSolverInterface> qpsolver_ptr_;          // qp solver for mpc
+  std::string output_interface_;                             // output command type
+
+  /* parameters for control*/
+  double ctrl_period_;              // deside control frequency
```

Comment:
can you please switch to [doxygen](http://wiki.ros.org/rosdoc_lite) comment formatting?